storage/remote/queue_manager.go

```diff
@@ -307,11 +353,78 @@ func NewQueueManager(
 	}
 
 	t.watcher = wal.NewWatcher(watcherMetrics, readerMetrics, logger, client.Name(), t, walDir)
+	if t.mcfg.Send {
+		t.metadataWatcher = NewMetadataWatcher(logger, sm, client.Name(), t, t.mcfg.SendInterval, flushDeadline)
+	}
 	t.shards = t.newShards()
 
 	return t
 }
 
+// AppendMetadata queues metadata to be sent to the remote storage. Metadata is sent all at once and is not parallelized.
```

Comment:
Hmm, I'm not sure how I feel about the comment `queues metadata to be sent...` since we actually are doing the send here. But also, that's just an implementation detail of the current implementation and wouldn't necessarily be the case in the future. :|

Reply:
Would `AppendMetadata sends metadata to the remote storage.` work? Keeping it simple...

Reply:
I'm okay with leaving it as is or making a change similar to your suggestion. I think the whole `Append` function may change to accept more data types in the future so the exact comment on this specific function, whose functionality might move into another area in the future, isn't super important to me.